packages/mui-joy/src/Menu/Menu.tsx

```diff
@@ -153,6 +153,26 @@ const Menu = React.forwardRef(function Menu(inProps, ref: React.ForwardedRef<HTM
     [modifiersProp],
   );
 
+  if (anchorEl) {
+    let ariaControls = null;
+    if ('getAttribute' in anchorEl) {
+      ariaControls = anchorEl.getAttribute('aria-controls');
+    } else if (typeof anchorEl === 'function') {
```

Comment:
Is it possible for the `anchorEl` to be a function? I thought that it will always be a DOM element.

Reply:
according to `anchorEl` prop description it is possible.

`An HTML element, [virtualElement], or a function that returns either. It's used to set the position of the popper. The return value will passed as the reference object of the Popper instance.`

Reply:
You are right!